backend/pom.xml

```diff
@@ -45,6 +45,18 @@
             <artifactId>jasypt-spring-boot-starter</artifactId>
             <version>3.0.3</version>
         </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+            <version>4.0.1</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
For `test` dependencies I'm less particular, but in some cases I would request using what's already available if possible. In this case that would be [`org.springframework.boot.test.web.client.TestRestTemplate`](https://www.baeldung.com/spring-boot-testresttemplate). Here I'm mentioning it just so you're aware.

If we use the "standard" libraries, be they Spring or Core Java:

* It compiles a touch faster.
* File size is a bit smaller.
* People are likely to be familiar with the libraries you're using.
* Fewer dependencies is arguably less complex.
* Long term support is more likely.

I know these bullet points are one-sided; there are many reasons to reach for the best library for the job.

I'm not asking for changes here. Just sharing my thoughts.